src/bq_data_collector.py

```diff
@@ -0,0 +1,244 @@
+import itertools
+import logging
+import os
+import warnings
+from typing import List, Dict
+
+import arrow
+import daiquiri
+import pandas as pd
+
+import src.utils.cloud_constants as cc
+from src.utils import bq_client_helper
+
+warnings.simplefilter(action='ignore', category=FutureWarning)
+warnings.simplefilter(action='ignore', category=Warning)
+
+daiquiri.setup(level=logging.INFO)
+_logger = daiquiri.getLogger(__name__)
+
+
+class BigQueryDataCollector:
+    def __init__(self, repos: List[str], bq_credentials_path: str = '', days: int = 3):
+        self._bq_client = BigQueryDataCollector._get_bq_client(bq_credentials_path)
+        self._golang_repo = bq_client_helper.get_gokube_trackable_repos(repo_dir=cc.GOKUBE_REPO_LIST)
+        self._knative_repo = bq_client_helper.get_gokube_trackable_repos(repo_dir=cc.KNATIVE_REPO_LIST)
+        self._kubevirt_repo = bq_client_helper.get_gokube_trackable_repos(repo_dir=cc.KUBEVIRT_REPO_LIST)
+        self._init_query_param(repos, days)
```

Comment:
if you go ahead with CSV, this won't scale better if we wanna track more other ecosystems. It is better to use dictionary/array based approach to keep details instead of dedicated instance variable for each ecosystem.

Reply:
Added dictionary with all eco-system and repo names.